src/chain/chain.rs

```diff
@@ -263,6 +276,7 @@ impl Chain {
                         .collect(),
                 };
 
+                self.process_churn = false;
```

Comment:
Should we only flip this to `false` when the section info is for our own section?

Reply:
Yes, I'll push the updated branch if you are looking at it.

Reply:
Sure. Or if you want me to wait with the review a bit, let me know.